final/scripts/app.js

```diff
@@ -64,6 +77,9 @@
     var selected = select.options[select.selectedIndex];
     var key = selected.value;
     var label = selected.textContent;
+    if (!app.selectedCities) {
```

Comment:
This unfortunately introduces another edge case. If you load the app, then never add any new cities, you'll ALWAYS get the original data unless you manually use the update button.